### ISSUE TITLE
REJECT shipment is not working on the WHS Dashboard

### ISSUE THREAD

**Issue by joaoluis-pdm** [PharmaLedger-IMI/fgt-workspace#83]
When shipment in status DELIVERED is rejected in the Dashboard of a WHS,... does not work.

When the shipment is in status RECEIVED , the error is different.

This is part of the "Unhappy path" of the Stage 4 tests.

**Comment by joaoluis-pdm** [comment]
The REST request is
```json
PUT https://api-whs2-fgt-dev.pharmaledger.pdmfc.com/traceability/shipment/update/MAH136366355-133
Host: api-whs2-fgt-dev.pharmaledger.pdmfc.com
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:100.0) Gecko/20100101 Firefox/100.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://whs2-fgt-dev.pharmaledger.pdmfc.com/
authorization: Basic ...redacted....=
Content-Type: text/plain;charset=UTF-8
Origin: https://whs2-fgt-dev.pharmaledger.pdmfc.com
Content-Length: 37
Connection: keep-alive
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-site
TE: trailers

{"status":"rejected","extraInfo":"r"}
```

The response is
```json
{"status":400,"error":"Bad Request","message":"Requester is not able to update to status: rejected."}
```

**Comment by joaoluis-pdm** [comment]
@TiagoV-PDMFC seems that the fix would be:
```js
diff --git a/fgt-api/model/SimpleShipment.js b/fgt-api/model/SimpleShipment.js
index 00f544ba..7d2f27b5 100644
--- a/fgt-api/model/SimpleShipment.js
+++ b/fgt-api/model/SimpleShipment.js
@@ -76,7 +76,7 @@ class SimpleShipment {
     }
 
     allowedRequesterStatusUpdate() {
-        return [ShipmentStatus.RECEIVED, ShipmentStatus.CONFIRMED].includes(this.status.status);
+        return [ShipmentStatus.DELIVERED, ShipmentStatus.RECEIVED, ShipmentStatus.CONFIRMED].includes(this.status.status);
     }
 }
```
(But it does not explain yet, the error when the status is RECEIVED...).

**Comment by joaoluis-pdm** [comment]
For the RECEIVED status, no REST service invocation appears on the F12 Network call, so the error might be in the frontend...

**Comment by joaoluis-pdm** [comment]
@TiagoV-PDMFC do you have an opinion on the fixes proposed in the previous post ?

**Comment by TiagoV-PDMFC** [comment]
I think the fix should be on the Shipment:

```
static getAllowedStatusUpdates(status){
        switch(status){
            case ShipmentStatus.CREATED:
                return [ShipmentStatus.REJECTED, ShipmentStatus.ON_HOLD, ShipmentStatus.PICKUP]
            case ShipmentStatus.ON_HOLD:
                return [ShipmentStatus.PICKUP, ShipmentStatus.REJECTED]
            case ShipmentStatus.PICKUP:
                return [ShipmentStatus.ON_HOLD, ShipmentStatus.REJECTED, ShipmentStatus.TRANSIT]
            case ShipmentStatus.TRANSIT:
                return [ShipmentStatus.REJECTED, ShipmentStatus.ON_HOLD, ShipmentStatus.DELIVERED]
            case ShipmentStatus.DELIVERED:
                return [ShipmentStatus.REJECTED, ShipmentStatus.RECEIVED]
            case ShipmentStatus.RECEIVED:
                return [ShipmentStatus.CONFIRMED]
            default:
                return [];
        }
    }```

**Comment by joaoluis-pdm** [comment]
@pccosta-pdm some change is still needed.

![image](https://user-images.githubusercontent.com/77448470/176730520-0760d8ab-95f3-4d11-8276-d408c938c4f3.png)

On the WHS0001. after performing "RECEIVE" on an incoming shipment,
if you press "REJECT" and "CONFIRM", nothing happens (and there is an error in the console):
![image](https://user-images.githubusercontent.com/77448470/176730860-8be32e39-1d6f-49cf-816d-9d51156b10b2.png)

(Do something similar to 7bfd05977d1f87e0be26774431c14c69c826ea10 ?)

**Comment by joaoluis-pdm** [comment]
E agora @pccosta-pdm / @TiagoV-PDMFC  temos um dilema
Dá erro

![imagem](https://user-images.githubusercontent.com/77448470/176956313-cca5ebb5-8169-4e96-b11f-2aa7d9f9cae2.png)


quando a documentação diz que pode ser rejeitada

<img width="882" alt="imagem" src="https://user-images.githubusercontent.com/77448470/176956490-c7fe35a9-9f6a-4210-9b14-7a2e08cb4a1f.png">


Qual é que está errada ?

**Comment by pccosta-pdm** [comment]
A documentação está correta @joaoluis-pdm, o fix foi feito em [d17529a](https://github.com/PharmaLedger-IMI/fgt-workspace/commit/d17529aad7386e97e8dc0c5363134aa77a1ac79b)
